impl/src/main/java/org/apache/myfaces/renderkit/html/util/HtmlRendererUtils.java

```diff
@@ -292,6 +292,12 @@ public static void decodeUISelectOne(FacesContext facesContext, UIComponent comp
                                 callback);
                     }
                 }
+                else 
+                {
+                    // means input was not submitted. set to empty string so we can validate required fields
+                    // if not set, a null value will skip validation -- see beginning of UIInput#validate
+                    ((EditableValueHolder)component).setSubmittedValue(RendererUtils.EMPTY_STRING);
```

Comment:
Setting to RendererUtils.EMPTY_STRING, similar as to Line 322 & 245

Not a fan of using an empty string here ( since it's latter converted to a null), maybe an enum would be better...?

Reply:
Not sure if enums are spec allowed due to this isEmpty() definition: 
https://jakarta.ee/specifications/faces/4.0/apidocs/jakarta/faces/component/uiinput#isEmpty(java.lang.Object)
```
If the value is null, return true. If the value is a String and it is the empty string, return true. If the value is an array and the array length is 0, return true. If the value is a List and the List is empty, return true. If the value is a Collection and the Collection is empty, return true. If the value is a Map and the Map is empty, return true. In all other cases, return false. 
```

I think we will need to keep it as is (set to empty string when no value submitted, if kept as null it would cause validation to be skipped)